src/metrics/include/metrics/operation.hpp

```diff
@@ -123,13 +147,24 @@ class OperationImpl final : private boost::noncopyable {
         return _events;
     }
 
-    void reportAt(time_point finished, OperationEvent<ClockSource>&& event) {
+    void reportAt(time_point started, time_point finished, OperationEvent<ClockSource>&& event) {
+        if (_threshold) {
+            _threshold->counter.total++;
+            if ((finished - started) > _threshold->duration) {
+                _threshold->counter.failed++;
+            }
+
+            if (_threshold->counter.failedPercentage() > _threshold->failedPercentageThreshold) {
+                throw OperationThresholdExceededException();
```

Comment:
Do I understand correctly that the threshold is checked every time an operation is reported?  Could this be problematic when operations first start running? For any threshold, if the first operation exceeds the `maxDuration` it would fail. I think it would make sense to gather more data before checked if the failPercentage is too great, to prevent giving a huge weight to the first operations.

Reply:
👍